### PR TITLE
Use default configuration when failed to read from file.

### PR DESCRIPTION
**Pull request by chenhao94**

### CODE REVIEW

File: src/config/config.cc [hunk 1]
```diff
@@ -59,10 +60,13 @@ std::shared_ptr<ConfigBase> ConfigFile::RegisterOrGet(
 }
 
 void ConfigFile::InitConfigJsonContext() {
-    if (const auto load_err = simdjson::padded_string::load(filepath_).get(json_context_.buf_)) {
-        LOG(ERROR) << __func__ << ": Failed to load Config: " << filepath_ << " "
-                   << simdjson::simdjson_error(load_err).what();
-        return;
+    if (filepath_.empty()) {
+        LOG(WARNING) << __func__ << ": Empty config path, using default configuration.";
+        json_context_.buf_ = simdjson::padded_string(std::string_view("{}"));
```
Comment (zhouxindong) [inline src/config/config.cc:65]: padded_string()的参数直接"{}"可以吗？或者"{}"_padded;

Comment (chenhao94) [reply src/config/config.cc:65]: 直接"{}"不可以，因为有const char*和sv两种歧义。用literal应该行但我不想using namespace

Comment (zhouxindong) [reply src/config/config.cc:65]: 明白了

**Review by zhouxindong — APPROVED** [review]

File: src/config/config.cc [hunk 2]
```diff
@@ -59,10 +60,13 @@ std::shared_ptr<ConfigBase> ConfigFile::RegisterOrGet(
...
+    } else if (const auto load_err = simdjson::padded_string::load(filepath_).get(json_context_.buf_)) {
+        LOG(ERROR) << __func__ << ": Failed to load Config \"" << filepath_
```
Comment (xkszltl) [inline src/config/config.cc:67]: config, lower case

File: tests/config_test.cc [hunk 3]
```diff
@@ -160,4 +160,13 @@ TEST_F(ConfigTest, ConfigDataGetStringRep) {
     EXPECT_FALSE(impl::ConfigDataGetStringRep(SomeRandomClass()).empty());
 }
 
+TEST_F(ConfigTest, WrongFilePath) {
+    ConfigFile empty_config("");
+    ConfigFile invalid_file_config("nonexistent.config");
```
Comment (xkszltl) [inline tests/config_test.cc:165]: .cfg or  .conf?
Also all of them are generally not a good extra because ext should describe format, not use case

**Review by xkszltl — APPROVED** [review]

**Pull request merged**